impeller/entity/entity_pass.h

```diff
@@ -52,6 +52,13 @@ class EntityPass {
 
   void IterateAllEntities(const std::function<bool(Entity&)>& iterator);
 
+  /// @brief Iterates all entities on this pass, returning if there is a
+  /// subpass.
+  bool IterateAllFlatEntities(const std::function<bool(Entity&)>& iterator);
```

Comment:
This is sort of an unnecessary restriction, but I want to avoid traversing the entire tree. Ideally if all children accepted opacity and there was a subpass, we could check if the subpass accepts opacity and just push it one layer down.

This would help cases where we've nested FadeTransitions/Opacity